test/integration/provider/rti.go

```diff
@@ -0,0 +1,86 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package provider
+
+import (
+	"fmt"
+
+	"github.com/gardener/machine-controller-manager-provider-openstack/pkg/client"
+	v1alpha1 "github.com/gardener/machine-controller-manager/pkg/apis/machine/v1alpha1"
+)
+
+// ITResourceTagKey is specifically used for integration test
+// primarily to avoid orphan collection of resources when the control cluster is
+// non seed cluster
+var ITResourceTagKey = "kubernetes.io-role-integration-test"
+
+// ResourcesTrackerImpl type keeps a note of resources which are initialized in MCM IT suite and are used in provider IT
+type ResourcesTrackerImpl struct {
+	MachineClass *v1alpha1.MachineClass
+	SecretData   map[string][]byte
+	ClusterName  string
+}
+
+// InitializeResourcesTracker initializes the type ResourcesTrackerImpl variable and tries
+// to delete the orphan resources present before the actual IT runs.
+func (r *ResourcesTrackerImpl) InitializeResourcesTracker(machineClass *v1alpha1.MachineClass, secretData map[string][]byte, clusterName string) error {
```

Comment:
In reference to my comment at L31
```suggestion
func (r *ResourcesTrackerImpl) InitializeResourcesTracker(machineClass *v1alpha1.MachineClass, secretData map[string][]byte, _ string) error {
```